services/users/service.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package users
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/shared/i18n"
+	"github.com/mattermost/mattermost-server/v5/shared/mlog"
+	"github.com/mattermost/mattermost-server/v5/store"
+)
+
+type ConfigFn func() *model.Config
+
+type UserService struct {
+	store  store.UserStore
+	config ConfigFn
+}
+
+func New(s store.UserStore, cfgFn ConfigFn) *UserService {
+	return &UserService{
+		store:  s,
+		config: cfgFn,
+	}
+}
+
+// CreateUser creates a user
+func (us *UserService) CreateUser(user *model.User) (*model.User, *model.AppError) {
+	// TODO: Are guest users are specific to chat or they are applicable for the whole suite?
```

Comment:
They should be applicable to the whole suite. But they may not be used by IC/FB right now.